### PR TITLE
Add Status subresource project idea

### PR DESCRIPTION
**Pull request by ArthurSens**
I'm adding everyone as a reviewer just to make sure I'm adding the right mentors to the right projects.

Please speak up if you're added to a project you didn't want to mentor (or vice-versa)

### CODE REVIEW

File: mentoring/gsoc/2024/project_ideas.md [hunk 1]
```diff
@@ -26,3 +26,13 @@ If you are a Prometheus-Operator team member and consider mentoring during the G
 ---
 
 ## Ideas
+
+### `*Monitor` status subresource
+
+- Description: Currently, there are 4 CRDs fully dedicated to configure Prometheus scrape targets, `PodMonitor`, `ServiceMonitor`, `Probes` and `ScrapeConfig`, and all of them are missing the `.Status` subresource. This project is about implementing the status subresource for all 4 CRDs to increase visibility and debuggability of scrape configuration. The fields added to the status subresource will be discused deeply during the mentorship, but a few ideas that already came up is showing list of Prometheus CRs that selected each Monitor CR and how many targets the resource identifies.
```
Comment (simonpasquier) [inline mentoring/gsoc/2024/project_ideas.md:32]:
```suggestion
- Description: Currently, there are 4 Custom Resource Definitions (CRDs) fully dedicated to configure Prometheus scrape targets, `PodMonitor`, `ServiceMonitor`, `Probes` and `ScrapeConfig`, and all of them are missing the `.Status` subresource. This project is about implementing the status subresource for all 4 CRDs to increase visibility and debuggability of scrape configuration. The fields added to the status subresource will be discussed deeply during the mentorship, but a potential idea is to list the Prometheus/PrometheusAgent objects that selected each Monitor object and how many targets are being scraped.
```

File: mentoring/gsoc/2024/project_ideas.md [hunk 2]
```diff
@@ -26,3 +26,13 @@ If you are a Prometheus-Operator team member and consider mentoring during the G
 ---
 
 ## Ideas
+
+### `*Monitor` status subresource
+
+- Description: Currently, there are 4 Custom Resource Definitions (CRDs) fully dedicated to configure Prometheus scrape targets, `PodMonitor`, `ServiceMonitor`, `Probes` and `ScrapeConfig`, and all of them are missing the `.Status` subresource. This project is about implementing the status subresource for all 4 CRDs to increase visibility and debuggability of scrape configuration. The fields added to the status subresource will be discussed deeply during the mentorship, but a potential idea is to list the Prometheus/PrometheusAgent objects that selected each Monitor object and how many targets are being scraped.
+- Recommended Skills: Go, Kubernetes, Kubernetes CRDs.
```
Comment (simonpasquier) [inline mentoring/gsoc/2024/project_ideas.md:33]: (nit) familiarity with Kubernetes is good enough for me
```suggestion
- Recommended Skills: Go, Kubernetes.
```

File: mentoring/gsoc/2024/project_ideas.md [hunk 3]
```diff
@@ -26,3 +26,13 @@ If you are a Prometheus-Operator team member and consider mentoring during the G
...
+- Expected project size: medium
```
Comment (simonpasquier) [inline mentoring/gsoc/2024/project_ideas.md:34]: I'd go with "large", like for the Daemonset topic.

Comment (ArthurSens) [reply mentoring/gsoc/2024/project_ideas.md:34]: Yeah, I was a bit in doubt with this one, but let's be safe and classify as big

**Pull request merged**